src/models.ts

```diff
@@ -346,3 +346,15 @@ export interface ISelection {
   regions: IIdentityValue<IEqualsDataReference | IBetweenDataReference>[];
   filters: (IBasicFilter | IAdvancedFilter)[];
 }
+
+export enum Permissions {
```

Comment:
We might incorporate permissions into ViewMode setting as it basically controls the view mode capabilities the report loads with.